src/context/CartContext.js

```diff
@@ -0,0 +1,136 @@
+      newCart.products = [...newCart.products, actions.product];
+      newCart.total = getTotalPrice(newCart);
+      return newCart;
+
+    case ACTIONS.EDIT:
+      newCart.products = newCart.products.map((product) => {
+        if (product.id === actions.product.id) {
+          return { ...product, quantity: actions.product.quantity };
+        }
+        return product;
+      });
+      newCart.total = getTotalPrice(newCart);
+      return newCart;
+
+    case ACTIONS.DELETE:
+      newCart.products = cart.products.filter(
+        (product) => product.id !== actions.productId
+      );
+      newCart.total = getTotalPrice(newCart);
+      return newCart;
+
+    case ACTIONS.INCREASE:
+      newCart.products = newCart.products.map((product) => {
+        if (product.id === actions.productId) {
+          return { ...product, quantity: product.quantity + 1 };
+        }
+        return product;
+      });
+      newCart.total = getTotalPrice(newCart);
+      return newCart;
+
+    case ACTIONS.DECREASE:
+      newCart.products = newCart.products.map((product) => {
+        if (product.id === actions.productId) {
+          return { ...product, quantity: product.quantity - 1 };
+        }
+        return product;
+      });
+      newCart.total = getTotalPrice(newCart);
+      return newCart;
+
+    default:
+      return newCart;
+  }
+};
+
+const initializer = (key) => {
+  if (!localStorage.getItem(key)) return { products: [], total: 0 };
+  return JSON.parse(localStorage.getItem(key));
+};
+
+const CartContext = createContext();
+
+export const CartContextProvider = ({ children }) => {
+  const [cart, dispatchCart] = useReducer(reducer, initializer('cart'));
+
+  useEffect(() => {
+    localStorage.setItem('cart', JSON.stringify(cart));
+  }, [cart]);
+
+  const addProduct = (product, quantity) => {
+    const newProduct = {
+      id: product.id,
+      name: product.data.name,
+      price: product.data.price,
+      image: product.data.mainimage.url,
+      stock: product.data.stock,
+      quantity: quantity,
+    };
+    const productInCart = cart.products.find((prod) => prod.id === product.id);
+    const productObject = {
+      type: ACTIONS.ADD,
+      product: newProduct,
+    };
+    if (productInCart) productObject.type = ACTIONS.EDIT;
+
+    dispatchCart(productObject);
+  };
+  const increase = (productId) => {
+    const productObject = {
+      type: ACTIONS.INCREASE,
+      productId,
+    };
+    dispatchCart(productObject);
+  };
+  const decrease = (productId) => {
+    const productObject = {
+      type: ACTIONS.DECREASE,
+      productId,
+    };
+    dispatchCart(productObject);
+  };
+
+  const removeProduct = (productId) => {
+    const productObject = {
+      type: ACTIONS.DELETE,
+      productId,
+    };
+    dispatchCart(productObject);
+  };
+
+  const context = {
+    addProduct: addProduct,
+    cart: cart,
+    removeProduct: removeProduct,
+    increase: increase,
+    decrease: decrease,
```

Comment:
En ES6 puedes pasar cualquier variable una sola vez y su identificador de objeto recibirá el mismo nombre, haciendo que estas líneas que escribiste, sean exactamente lo mismo que lo siguiente.

```suggestion
    addProduct,
    cart,
    removeProduct,
    increase,
    decrease,
```